### PR TITLE
snap: Fix snap build with kernel fragments

### PR DESCRIPTION
**Pull request by ganeshmaharaj**
Signed-off-by: Ganesh Maharaj Mahalingam <ganesh.mahalingam@intel.com>

### CODE REVIEW

**Comment by ganeshmaharaj** [comment]
@egernst @devimc @grahamwhaley

**Comment by ganeshmaharaj** [comment]
@egernst @devimc @chavafg I ran the test suite 4 times locally and i was not able to reproduce the issue. 

this is the output from the last run. 
```
[4] Checking CPU cgroups in the host
[4] /home/ganeshma/go/src/github.com/kata-containers/tests/integration/docker/cgroups_test.go:110
[4]   checking whether cgroups can be deleted
[4]   /home/ganeshma/go/src/github.com/kata-containers/tests/integration/docker/cgroups_test.go:130
[4]     with a running process
[4]     /home/ganeshma/go/src/github.com/kata-containers/tests/integration/docker/cgroups_test.go:131
[4]       should be deleted
[4]       /home/ganeshma/go/src/github.com/kata-containers/tests/integration/docker/cgroups_test.go:132
[4] ------------------------------
[4] docker exec check stdout forwarded using exec
[4]   should displayed it
[4]   /home/ganeshma/go/src/github.com/kata-containers/tests/integration/docker/exec_test.go:61
[4] Running command '/usr/bin/docker [docker run --cidfile /tmp/cid705408909/D7GDm1wdduJmjNUghwf6h8ysl00tF4 --runtime kata-runtime -td --name D7GDm1wdduJmjNUghwf6h8ysl00tF4 busybox sh]'
[6] [bash -c printf "g\nn\n\n\n\nw\n" | fdisk /tmp/dd689310252]
[6] Timeout: 120 seconds
[6] Exit Code: 0
[6] Stdout:
[6] Welcome to fdisk (util-linux 2.31.1).
```
Any thoughts on the next steps ?

**Comment by grahamwhaley** [comment]
/test
as what I see in the CI logs is a test fail, that I suspect could be sporadic, so let's re-trigger...

```
[1] • Failure [5.304 seconds]
[1] Checking CPU cgroups in the host
[1] /tmp/jenkins/workspace/kata-containers-packaging-ubuntu-18-04-snap-PR/go/src/github.com/kata-containers/tests/integration/docker/cgroups_test.go:110
[1]   checking hosts's cpu cgroups
[1]   /tmp/jenkins/workspace/kata-containers-packaging-ubuntu-18-04-snap-PR/go/src/github.com/kata-containers/tests/integration/docker/cgroups_test.go:191
[1]     container with cpu and cpuset constraints
[1]     /tmp/jenkins/workspace/kata-containers-packaging-ubuntu-18-04-snap-PR/go/src/github.com/kata-containers/tests/integration/docker/cgroups_test.go:192
[1]       shold have its cgroup set correctly [It]
[1]       /tmp/jenkins/workspace/kata-containers-packaging-ubuntu-18-04-snap-PR/go/src/github.com/kata-containers/tests/integration/docker/cgroups_test.go:193
[1] 
[1]       Expected error:
[1]           <*os.PathError | 0xc00010e5a0>: {
[1]               Op: "open",
[1]               Path: "/sys/fs/cgroup/cpu/docker/kata_8de29f587942cdfaab89789147dc32475d62e1b70be70b9b3f8e490afb9dd61f/cpu.shares",
[1]               Err: 0x2,
[1]           }
[1]           open /sys/fs/cgroup/cpu/docker/kata_8de29f587942cdfaab89789147dc32475d62e1b70be70b9b3f8e490afb9dd61f/cpu.shares: no such file or directory
[1]       not to have occurred
[1] 
[1]       /tmp/jenkins/workspace/kata-containers-packaging-ubuntu-18-04-snap-PR/go/src/github.com/kata-containers/tests/integration/docker/cgroups_test.go:203
[1] ------------------------------
[1] 
[1] 
[1] Summarizing 1 Failure:
[1] 
[1] [Fail] Checking CPU cgroups in the host checking hosts's cpu cgroups container with cpu and cpuset constraints [It] shold have its cgroup set correctly 
[1] /tmp/jenkins/workspace/kata-containers-packaging-ubuntu-18-04-snap-PR/go/src/github.com/kata-containers/tests/integration/docker/cgroups_test.go:203
[1] 
[1] Ran 12 of 13 Specs in 231.147 seconds
[1] FAIL! -- 11 Passed | 1 Failed | 0 Pending | 1 Skipped --- FAIL: TestIntegration (231.15s)
[1] FAIL
```

**Comment by devimc** [comment]
fixed snap CI , please take a look https://github.com/kata-containers/packaging/pull/441

**Comment by devimc** [comment]
@ganeshmaharaj  rebase  :smile:

**Comment by ganeshmaharaj** [comment]
@devimc done. :)

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -193,28 +193,33 @@ parts:
...
-        ;;
-
-        *)
-          echo "ERROR: Unsupported architecture $(arch)"
-          exit 1
-        ;;
-      esac
+      archfragdir="${SNAPCRAFT_STAGE}/kernel/configs/fragments/$(arch)"
+      if [ -f "${archfragdir}/.config" ]; then
+        config="fragments/$(arch)/.config"
```
Comment (devimc) [inline snap/snapcraft.yaml]: who is going to generate this file?

Comment (ganeshmaharaj) [reply snap/snapcraft.yaml]: The build script will generate this file on the fly and copy it into the location referenced above. This should be re-generated every time we run the test.

Comment (grahamwhaley) [reply snap/snapcraft.yaml]: @ganeshmaharaj - is this related to https://github.com/kata-containers/tests/pull/1490 as well?
If so, is there some dependence/sequence we need to land the relevant PRs in?

Comment (ganeshmaharaj) [reply snap/snapcraft.yaml]: Should not affect the stable branches for now @grahamwhaley. This should be alright to go standalone before kernel fragment patch.

**Comment by ganeshmaharaj** [comment]
Thanks @devimc for catching this. Closing this PR as @devimc has a cleaner fix for the same.

**Pull request closed without merging**